src/app/models/treeoperations.cpp

```diff
@@ -157,3 +157,19 @@ void TreeOperations::notifyDbWasUnloaded(int dbIndex)
 {
     emit closeDbKeys(m_connection, dbIndex);
 }
+
+void TreeOperations::deleteDbKey(int dbIndex, ConnectionsTree::KeyItem& key)
+{
+    RedisClient::Command::Callback callback = [this, dbIndex, &key](const RedisClient::Response&, const QString&)
```

Comment:
Second arg - is error message here, check it. If it is not empty then call callback and pass this error:

```c++
if (!error.isEmpty()) {
  callback(QString("Cannot remove key: %1").arg(error))
  return;
}

```